sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_parallel_for_pipeline_job_output_name_version.yaml

```diff
@@ -0,0 +1,17 @@
+display_name: parallel_for_pipeline
+type: pipeline
+jobs:
+  parallel_body:
+    type: command
+    inputs:
+      component_in_path:
+        type: uri_file
+        path: https://dprepdata.blob.core.windows.net/demo/Titanic.csv
+    component: ../components/helloworld_component.yml
```

Comment:
Add test cases for SDK/CLI when
1. Name version only configured in node level
2. only configured in pipeline level
3. configured in both node and pipeline level
Make sure we passed expected payload to server

Reply:
I add test_register_output_cli and test_register_output_sdk in pipeline_job/e2etests/test_pipeline_job.py